examples/coap-server.c

```diff
@@ -404,28 +442,29 @@ hnd_get_example_data(coap_context_t *ctx UNUSED_PARAM,
         coap_string_t *query,
         coap_pdu_t *response
 ) {
-  if (!example_data_ptr) {
+  if (!example_data_value) {
     /* Initialise for the first time */
     int i;
-    example_data_ptr = coap_new_binary(1500);
-    if (example_data_ptr) {
-      example_data_ptr->length = 1500;
+    coap_binary_t *value = coap_new_binary(1500);
+    if (value) {
+      value->length = 1500;
       for (i = 0; i < 1500; i++) {
         if ((i % 10) == 0) {
-          example_data_ptr->s[i] = 'a' + (i/10) % 26;
+          value->s[i] = 'a' + (i/10) % 26;
         }
         else {
-          example_data_ptr->s[i] = '0' + i%10;
+          value->s[i] = '0' + i%10;
         }
       }
     }
+    example_data_value = alloc_resource_data(value);
   }
   response->code = COAP_RESPONSE_CODE_CONTENT;
   coap_add_data_large_response(resource, session, request, response, token,
                                query, example_data_media_type, -1, 0,
-                               example_data_ptr ? example_data_ptr->length : 0,
-                               example_data_ptr ? example_data_ptr->s : NULL,
-                               NULL, NULL);
+                    example_data_value ? example_data_value->value->length : 0,
```

Comment:
Because of the way `alloc_resource_data()` works, `example_data_value->value` could be NULL here. This happens when `coap_new_binary()` in `hnd_get_example_data()` fails and returns NULL which is stored as `example_data_value->value` in line 460.

Reply:
Hmm.  If `coap_new_binary()` fails line 448, then on calling `alloc_resource_data()` line 460, this will return NULL line 184.

The intention was that `alloc_resource_data()` only returned something that was correctly formed.  Added to document `alloc_resource_data()`
````
/*
 * This will return a correctly formed transient_value_t *, or NULL.
 * If an error, the passed in coap_binary_t * will get deleted.
 */
````

Reply:
Ah, yes, you are right. Somehow, I have missed that `alloc_resource_data()` checks the passed parameter `value` before doing anything else.